shell/src/main/java/alluxio/cli/docgen/ConfigurationDocGenerator.java

```diff
@@ -206,17 +223,82 @@ public static void writeYMLFile(Collection<? extends PropertyKey> defaultKeys, S
 
   /**
    * Generates the configuration docs.
+   * @param validate validate flag
    */
-  public static void generate() throws IOException {
+  public static void generate(boolean validate) throws IOException {
     Collection<? extends PropertyKey> defaultKeys = PropertyKey.defaultKeys();
     defaultKeys.removeIf(key -> key.isHidden());
     String homeDir = new InstancedConfiguration(ConfigurationUtils.defaults())
         .get(PropertyKey.HOME);
     // generate CSV files
     String filePath = PathUtils.concatPath(homeDir, CSV_FILE_DIR);
-    writeCSVFile(defaultKeys, filePath);
+    writeCSVFile(defaultKeys, filePath, validate);
+
+    if (validate) {
+      boolean hasDiff = false;
+      String[] fileNames = {"user-configuration.csv", "master-configuration.csv",
+          "worker-configuration.csv", "security-configuration.csv",
+          "common-configuration.csv", "cluster-management-configuration.csv"};
+
+      for (String fileName : fileNames) {
+        String tempPrefix = "temp-";
+        String fileNameTemp = tempPrefix.concat(fileName);
+
+        // create the ProcessBuilder for the diff command
+        ProcessBuilder pb = new ProcessBuilder("diff", PathUtils.concatPath(filePath, fileName),
+            PathUtils.concatPath(filePath, fileNameTemp));
+        // redirect the diff output to a string
+        String output = IOUtils.toString(pb.start().getInputStream(), StandardCharsets.UTF_8);
+
+        if (output.length() > 0) {
+          System.out.println(output);
+          hasDiff = true;
+          System.out.println("Config file " + fileName + " changed.");
+        }
+        File f = new File(PathUtils.concatPath(filePath, fileName));
```

Comment:
this section renames the temp generated file into the actual generated file, but a validation should not touch any files.

on the other hand, the generated files do need to get deleted